thunder/torch/__init__.py

```diff
@@ -2001,6 +2001,11 @@ def amin(a, /, dim=None, keepdim: bool = False):
     )
 
 
+@torchsymbol(torch.clone, is_method=True)
+def clone(a: TensorProxy, *, memory_format=torch.preserve_format) -> TensorProxy:
+    return a
```

Comment:
Could that be `a + 0`?

Reply:
It could be `a+0` of course... tell me more about where you're coming from here, I'm not quite following the 'why'.

I actually wasn't at the triage meeting. @mruberry could you give more insight into #342's triage comment?

Reply:
`a+0` will be a copy. Returning just `a` will produce a tensor with the same storage.

Reply:
Good point. I addressed this with a cheeky `repeat` usage, which avoids the issue of an appropriate type for `0` s.t. this doesn't introduce a cast.

Please 'resolve' if this works for you.

Reply:
I am not sure this will preserve the memory format, though. It's not uncommon to see use cases with channels last and alike.

Reply:
AFAICT [the implementation of repeat](https://github.com/Lightning-AI/lightning-thunder/blob/54bb6146ff757905925f8d9ea2197870c4971011/thunder/torch/__init__.py#L919) does not do anything to reorder dimensions or similar. it's basically just a broadcast from one tensor to another. Given that the dimensions would be the same, I imagine this just ends up being a copy.

For other reasons I do think we want a `copy` primitive (which would be a more natural mapping, here), but I'm not sure others agree. Reasonable might argue that "broadcast to the same dimensions *is* the way to do copy", and I'm not sure I would disagree.

@t-vi thoughts on what we should do here?

Reply:
Sure, but then it works only if memory format is selected. Is it tested? What about different memory formats?

Reply:
Hey @nikitaved we had a chat and I missed something earlier:

if we just `return a` here without adding the `@torchsymbol` annotation then the clone won't actually appear in the trace, it'll just give back the same tensor. in triage review we noted the issue that this could result in weirdness if something modifies the return value of `clone`, but this actually falls under the general rubric of in-place ops and functionalizing the trace, and the sentiment was that we shouldn't try to boil the ocean in a PR that just adds an op.

Your point on memory formats is a good one though; for now I'll just add a conditional and check that preserve_format is used, so we at least get some notification if this is awry.